kadai1/shinji/convimg/convimg.go

```diff
@@ -0,0 +1,87 @@
+// 画像の変換機能を提供します。
+package convimg
+
+import (
+	"fmt"
+	"image"
+	"image/gif"
+	"image/jpeg"
+	"image/png"
+	"os"
+	"path/filepath"
+)
+
+type Ext string
+
+const (
+	PNG  Ext = ".png"
+	JPEG Ext = ".jpg"
+	GIF  Ext = ".gif"
+)
+
+// 拡張子を変更します。
+func convExt(srcPath string, to Ext) string {
+	ext := filepath.Ext(srcPath)
+	return srcPath[:len(srcPath)-len(ext)] + string(to)
+}
+
+// 画像ファイルを出力
+func outputImage(dst *os.File, img image.Image, to Ext) {
```

Comment:
errorを返して、呼び出し側でエラー時の処理を決められるようにすると使いやすいかも
```suggestion
func outputImage(dst *os.File, img image.Image, to Ext) error {
```

Reply:
ありがとうございます！
errorを返して、呼び出し側でエラー時の処理を決められる実装に修正しました。

https://github.com/gopherdojo/gopherdojo-studyroom/blob/98cb6b2d75b6eba1feef7d3477c6b556962cf63b/kadai1/shinji/convimg/convimg.go#L29

これってもしかして、このパッケージ全体でそうするのが適切なんでしょうか？
お時間ある時にコメントいただけますと幸いです。

Reply:
はい、その通りです。

一般的に、例外処理はライブラリの利用者に任せます。
利用者は再度ファイル選択のプロンプトを出すかもしれないし、何もしないかもしれないです。
ライブラリ内で処理する（例外を握りつぶす）と使いづらいです。

例: `f, err := os.Open("a.txt")`が、もし`f := os.Open("a.txt")`で例外が発生したらos.Stderrに出力する仕様だったら、めっちゃ使いづらいですよね....

Goの例外処理は、try-catchではなく、errorを返す書き方をします。
標準ライブラリに倣って、例外が発生する関数は、最後の戻り値をerrorにすることを推奨します。

Reply:
なるほど、ありがとうございます！
確かに、error処理をどうするかが利用者側で決められないのは使いにくいですね‥。
修正してみます！